### PR TITLE
Javascript javascript3 week1

### PR DESCRIPTION
**Pull request by Hailegebriel22**

### CODE REVIEW

File: javascript/javascript3/week1/gifsearch.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#validcheck{
+    margin-left: 25%;
+}
+#term {
+       margin-left: 25%;
+    width: 15%;
+padding: 1em 1em;
+}
+
+#mybtn {
```
Comment (wael85) [inline javascript/javascript3/week1/gifsearch.css]: befor saving any file try alt+shift+f to format the space and aligne your code

Comment (wael85) [reply javascript/javascript3/week1/gifsearch.css]: and avoid using (mybtn- myinput--) for example gif-search-btn it gives more meaning

File: javascript/javascript3/week1/gifsearch.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/* Create a site where a user can search for any word.
+ When searching a word the application will find a gif using the searched word 
+ using the giphy api: https://developers.giphy.com/docs/ 
+ Here is how it is going to work: The user can write some text indicating the gif he is looking for,
+ click a button and then a gif will be found (using the searched word) and the gif will be displayed to the user.
+Add an input element, where the user can specify how many gif results the user wants. */
+
+
+const gifDiv = document.getElementById("gifwrapper");
```
Comment (wael85) [inline javascript/javascript3/week1/gifsearch.js]: Remove unused code

File: javascript/javascript3/week1/gifsearch.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const ul = document.createElement('ul')
```
Comment (wael85) [inline javascript/javascript3/week1/gifsearch.js]: Rember ' ; ' it is necessary :-)

File: javascript/javascript3/week1/gifsearch.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const numberOfGifs = document.getElementById("numgifs")
+function searchGif() {
+    removeAllChildNodes(ul)
```
Comment (wael85) [inline javascript/javascript3/week1/gifsearch.js]: you can use ul.innerText = "";

File: javascript/javascript3/week1/gifsearch.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    }
+}
+function removeAllChildNodes(parent) {
+    while (parent.firstChild) {
+        parent.removeChild(parent.firstChild);
+    }
+}
+
+const mybtn = document.getElementById("mybtn")
```
Comment (wael85) [inline javascript/javascript3/week1/gifsearch.js]: you can use onclick 
`<button` id="mybtn" onclick="searchGif()">search for gif</button>`

File: javascript/javascript3/week1/gifsearch.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        metaImage.forEach(element => {
+            const li = document.createElement("li")
+            li.style.listStyle = "none";
+            const img = document.createElement("img")
+            img.src = element.fixed_width_downsampled.url;
+            li.appendChild(img)
+            ul.appendChild(li)
+
+            document.body.appendChild(ul)
```
Comment (wael85) [inline javascript/javascript3/week1/gifsearch.js]: document.body.appendChild(ul)
it should be out foreach

File: javascript/javascript3/week1/gifsearch.js (hunk 5 above)
Comment (Hailegebriel22) [reply javascript/javascript3/week1/gifsearch.js]: I can but mentors advised that it is better to use 'click' than  'onclick' on HTML.

**Pull request merged**